ios/FluentUI/Vnext/Drawer/SlideOverPanel.swift

```diff
@@ -103,13 +107,19 @@ struct MSFSlideOverPanel<Content: View>: View {
         guard backgroundDimmed else {
             return tokens.backgroundClearColor
         }
-        let opacity = resolvedPanelStateValue(tokens.backgroundClearOpacity, tokens.backgroundDimmedOpacity)
-        return tokens.backgroundDimmedColor.opacity(opacity)
-    }
 
-    private var shadowColor: Color {
-        let opacity = resolvedPanelStateValue(0.0, tokens.shadowOpacity)
-        return tokens.shadowColor.opacity(opacity)
+        // animate opacity
+        var opacity: Double = 0.0
+        switch transitionState {
+        case .collapsed:
+            opacity = 0
+        case .expanded:
+            opacity = 1
+        case .inTransisiton:
+            opacity = percentTransition ?? 0
+        }
+
+        return tokens.backgroundDimmedColor.opacity(opacity)
```

Comment:
Why aren't we animating from clear color to the background dimmed instead of using the opacity?
Is it because of the gesture scenario that calculates the percentage that reflects the opacity?

Reply:
yes, also swiftui only animates properties that are modified within `withAnimation` block